File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.kt
fun main() {
    val s = readln()
    val boom = readln()
    val boomLen = boom.length
    val answer = StringBuilder()

    for (c in s) {
        answer.append(c)

        if (answer.length >= boomLen && answer.substring(answer.length - boomLen, answer.length) == boom) {
            answer.delete(answer.length - boomLen, answer.length)
        }
    }

    if (answer.isEmpty()) println("FRULA")
    else println(answer.toString())
}

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.cc
#include <bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    string s, boom;
    cin >> s >> boom;
    string prevStr;
    int boomLen = boom.length();
    string answer;

    for(char c: s) {
        answer.push_back(c);

        if (answer.length() >= boomLen && answer.substr(answer.length() - boomLen, boomLen) == boom) {
            answer.erase(answer.size() - boomLen, boomLen);
        }
    }

    if (answer.empty()) cout << "FRULA" << '\n';
    else cout << answer << '\n';
}